### DISCUSSION TITLE
Should we ignore renaming Application class in new_project script?

### DISCUSSION THREAD

**Discussion by luongvo** [nimblehq/android-templates#357] [RFCs]
## Issue

This is the old logic, however, I don't think we should rename the Application class to fit the **generated project name**. There is no best practice to rename it. Should we name the Application class as `MainApplication` from the templates & samples? 💭 

IMO, removing `Renaming application class...` logic could simplify our script logic and allow us to focus on the rest of things more efficiently.

## Solution

Instead of naming the Application class as `TemplateXMLApplication` and `TemplateComposeApplication` in the templates, and then having logic in `new_project` script to find and rename that class to fit the generated project name (e.g. `SampleXmlApplication`) and its reference in the `AndroidManifest.xml` file, we could just name them as `MainApplication` from the template level and ignore renaming it in scripting level. The idea is to keep the default name as `MainApplication` and let the developer rename it later if it's really necessary to do.

- Rename the Application class in templates and samples to `MainApplication.kt` and their references in `AndroidManifest.xml`.
- Remove the `renameApplicationClass` logic in the `new_project` script. https://github.com/nimblehq/android-templates/blob/3924ed99f7c50c56b1980bbf81c0e4ddd684c692/scripts/new_project.kts#L240-L260

## Who Benefits?

Developers

## What's Next?

- [ ] Creating a chore ticket if everyone agrees to remove that logic.
- [ ] Creating a content ticket to update the project structure in Compass.

<img src="https://user-images.githubusercontent.com/16315358/204707836-d2fb0303-69d6-4bf5-88b4-74adff83a1d7.png" width=320 />

**Comment by toby-thanathip** [comment]
Voted ✅ 

I think it makes sense to remove it, as the application name doesn't bring much value anyways

**Comment by lydiasama** [comment]
Voted. 🙌🏻

**Comment by manh-t** [comment]
Voted ✅

**Comment by Wadeewee** [comment]
Voted ✅

**Comment by minhnimble** [comment]
Voted ✅

I would prefer going with less code and simplicity most of the time so this cleanup makes sense to me.

**Comment by hoangnguyen92dn** [comment]
@luongvo Would we stick with `MainApplication` for the Application class in all our next projects? 🤔 If so, I would also suggest adding one more action by updating the project structure on the compass. What do you think? 😄 
![image](https://user-images.githubusercontent.com/6950766/204706228-6d359cc8-40d5-430a-830e-7685e93c63cc.png)

**Reply by luongvo** [reply]
@hoangnguyen92dn Wait, do you mean you agree to name it to default as `MainApplication` or `MainApplicationClass`?

> Would we stick with MainApplication for the Application class in all our next projects?

Yes, I think we should. I see some of the projects named `MyApplication`, `MyApp`, or simply `App`. Naming it as `MainApplication` is also a good option.

> If so, I would also suggest adding one more action by updating the project structure on the compass. What do you think? 😄

If you mean using `MainApplication` instead of `MainApplicationClass` so Yes, we should make an update too 🙏

**Reply by hoangnguyen92dn** [reply]
@luongvo I agree updating to `MainApplication.kt`

**Reply by luongvo** [reply]
@hoangnguyen92dn I added 1 more action to update the Compass content, please vote 🤗

**Reply by hoangnguyen92dn** [reply]
@luongvo Just have an idea in mind, how about adding an option to change the Application class if developers want to do so? 🤔 By default, it will be MainApplication.kt

**Reply by luongvo** [reply]
@hoangnguyen92dn I still think we don't need to support it:
- The application class name does not bring much value to the project. Hence, naming it at the script level does not bring much sense to the project structure.
- The developer could rename it easily after generating the project. That's not a hard action (when compared to the project package name for example) so automating does not help us to solve a complex issue here.

**Reply by hoangnguyen92dn** [reply]
@luongvo Yeah. So that's fine if we stick with the `MainApplication.kt`

**Comment by doannimble** [comment]
Voted, we actually don't need a specific application name.

**Comment by toby-thanathip** [comment]
@luongvo Looks like it's settled! What happened to the "Convert discussion to issue" button? 💭

**Reply by toby-thanathip** [reply]
I don't think it's possible yet for RFC discussions 🙈

**Comment by toby-thanathip** [comment]
Issue created: https://github.com/nimblehq/android-templates/issues/361
